### PR TITLE
Wrm gradebook final

### PR DESCRIPTION
**Pull request by camliu89**
Adding all the logics to the gradebook.

### CODE REVIEW

File: tutor/src/screens/gradebook/aggregate-result-cell.js [hunk 1]
```diff
@@ -1,16 +1,31 @@
...
 const Cell = getCell('0 10px');
 
+const getPoints = (tasks) => {
+  const aggregatePoints = sumBy(tasks, (t) => t.correct_exercise_count);
+  return S.numberWithOneDecimalPlace(aggregatePoints/tasks.length);
+};
+
+const getPercentage = (data) => {
+  return data ? `${S.asPercent(data.average_score)}%` : '100';
+};
```
Comment (nathanstitt) [inline tutor/src/screens/gradebook/aggregate-result-cell.js:17]: how about adding these to the task model?  that way the'd be avail to other code that might need them

File: tutor/src/screens/gradebook/student-data-sorter.js [hunk 2]
```diff
@@ -2,10 +2,12 @@ import { isNumber } from 'lodash';
 
 const percent = (num, total) => Math.round((num / total) * 100) || 0;
 
-const getSortValue = function(student, index, dataType, displayAs) {
-
-  if (!isNumber(index)) { return (student.last_name || student.name).toLowerCase(); }
-  const task = student.data[index];
+const getSortValue = function(student, key, dataType, displayAs) {
+  //name, course_average, homework_score, reading_score
```
Comment (nathanstitt) [inline tutor/src/screens/gradebook/student-data-sorter.js:6]: 😁 left in by mistake I presume?

File: tutor/src/screens/gradebook/task-result-cell.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import { React, styled, useObserver } from 'vendor';
+import { observer } from 'mobx-react';
```
Comment (nathanstitt) [inline tutor/src/screens/gradebook/task-result-cell.js:2]: observer is exported by `vendor` - it's in tutor/src/vendor and should export pretty much everything

File: tutor/src/screens/gradebook/student-data-sorter.js (hunk 2 above)
Comment (camliu89) [reply tutor/src/screens/gradebook/student-data-sorter.js:6]: oh no, I just added this to know what the keys might be set to. I should make this comment more readable.

**Pull request merged**